added createdAt timestamp to pending mentorship requests tab

rickywid opened the pull request:
Tasks performed:
- add createdAt timestamp to Pending mentorships tab for both user and mentor user types.

Comments:
Maybe we can add a 'Received: (timestamp)' for Mentors and add a 'Sent: (timestamp)' for Users. If we just display the timestamp on it's own it may seem ambiguous to some?

roymoran commented on src/components/mentor/mentorshipsList.tsx:
```diff
@@ -271,6 +272,7 @@ class MentorshipsList extends React.Component<
                 displayOptions={displayOptions}
                 handleCheckClick={this.handleCheckClick}
                 handleCommentClick={this.handleCommentClick}
+                createdAt={TimeFormattingHelper.getTimeAgo(unixTime)}
```
We can simplify this a little. You can avoid passing down createdAt as an additional prop since we are already passing mentorship down to and it contains the `createdAt` property.

roymoran commented on src/components/shared/contentListItem.tsx:
```diff
@@ -20,6 +19,7 @@ interface IContentListItemProps {
   profileId?: string;
   userId?: string;
   userType: string;
+  createdAt?: string;
```
remove prop additional prop

roymoran commented on src/components/shared/contentListItem.tsx:
```diff
@@ -158,7 +166,7 @@ class ContentListItem extends React.Component<TProps, IContentListItemState> {
           </OptionsWrapper>
         ) : (
-          ''
+          <CreatedAt>{createdAt}</CreatedAt>
```
```suggestion
          <CreatedAt>{TimeFormattingHelper.getTimeAgo(unixTime)}</CreatedAt>
```

roymoran replied:
Don't use the `getTimeAgo` function. Create another function that displays the date as shown in the XD file `Tue 6 Nov`

roymoran commented on src/components/user/mentorshipsList.tsx:
```diff
@@ -263,9 +264,9 @@ class MentorshipsList extends React.Component<
       return (
         <List>
-          {mentorships.map(mentorship => {
+          {mentorships.map((mentorship: any) => {
```
Remove `: any`

roymoran commented on src/components/user/mentorshipsList.tsx:
```diff
@@ -263,9 +264,9 @@ class MentorshipsList extends React.Component<
+          {mentorships.map((mentorship: any) => {
             const profileId = mentorship.mentorId;
-
+            const unixTime = new Date(mentorship.createdAt).getTime();
```
remove this line

roymoran commented on src/components/user/mentorshipsList.tsx:
```diff
@@ -277,6 +278,7 @@ class MentorshipsList extends React.Component<
                 displayOptions={displayOptions}
                 handleCheckClick={this.handleCheckClick}
                 handleCommentClick={this.handleCommentClick}
+                createdAt={TimeFormattingHelper.getTimeAgo(unixTime)}
```
remove this prop

roymoran commented on src/components/shared/models/mentorship.ts:
```diff
@@ -21,6 +21,8 @@ export class Mentorship {
   public Status?: string;
   public UserMessage?: string;
+  public createdAt?: string;
+  public updatedAt?: string;
```
Change the casing here to follow convention

```suggestion
  public CreatedAt?: string;
  public UpdatedAt?: string;
```

roymoran replied:
Questions, why did you include UpdateAt? This isn't currently part of the response.

roymoran commented on src/components/shared/contentListItem.tsx:
```diff
@@ -116,7 +124,7 @@ class ContentListItem extends React.Component<TProps, IContentListItemState> {
   public render() {
     const { optionsSelected } = this.state;
-    const { mentorship, displayOptions, profileId, userId, userType } = this.props;
+    const { mentorship, displayOptions, profileId, userId, userType, createdAt } = this.props;
```
remove additional prop

roymoran commented on src/components/mentor/mentorshipsList.tsx:
```diff
@@ -257,9 +258,9 @@ class MentorshipsList extends React.Component<
       return (
         <List>
-          {mentorships.map(mentorship => {
+          {mentorships.map((mentorship: any) => {
```
remove `: any`

roymoran commented on src/components/mentor/mentorshipsList.tsx:
```diff
@@ -257,9 +258,9 @@ class MentorshipsList extends React.Component<
+          {mentorships.map((mentorship: any) => {
             const profileId = mentorship.userId;
-
+            const unixTime = new Date(mentorship.createdAt).getTime();
```
remove this line

roymoran commented on src/components/shared/models/mentorship.ts:
```diff
@@ -30,5 +32,7 @@ export class Mentorship {
     this.Status = mentorship.status;
     this.UserMessage = mentorship.userMessage;
+    this.createdAt = mentorship.createdAt;
+    this.updatedAt = mentorship.updatedAt;
```
Let's remove `updatedAt` for now since we aren't receiving this as part of the api response

rickywid commented:
- add new TimeFormattingHelper function
- removed updatedAt property from Mentorship model
- removed createdAt prop
- add timestamp to pending requests as per XD design
